### PR TITLE
[#25] Change order of arguments

### PR DESCRIPTION
**Pull request by RumyantsevMichael**
I have changed an order of parameters according to #25. Also, I've added a d.ts file right in the repo, so once the PR is approved it would be great to deprecate the [existing typings](https://www.npmjs.com/package/@types/reduce-reducers) as described [here](https://github.com/DefinitelyTyped/DefinitelyTyped#removing-a-package).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,41 +1,41 @@
...
-#### What is the difference between `reduceReducers` and `combineReducers`?
-
-This StackOverflow post explains it very well: https://stackoverflow.com/a/44371190/5741172
+# reduce-reducers
+
+[![Build Status](https://travis-ci.org/redux-utilities/reduce-reducers.svg?branch=master)](https://travis-ci.org/redux-utilities/reduce-reducers)

+[![npm Version](https://img.shields.io/npm/v/reduce-reducers.svg)](https://www.npmjs.com/package/reduce-reducers)

+[![npm Downloads Monthly](https://img.shields.io/npm/dm/reduce-reducers.svg)](https://www.npmjs.com/package/reduce-reducers)
```
Comment (timche) [inline README.md]: The badges are somehow now each in a separate line. Please revert them being all inline please.

File: README.md [hunk 2]
```diff
@@ -1,41 +1,41 @@
...
+
+## FAQ
+
+#### Why?
+
+Originally created to combine multiple Redux reducers that correspond to different actions (e.g. [like this](https://github.com/acdlite/redux-fsa/blob/master/src/handleActions.js#L12)). Technically works with any reducer, not just with Redux, though I don't know of any other use cases.
+
+#### What is the difference between `reduceReducers` and `combineReducers`?
+
+This StackOverflow post explains it very well: https://stackoverflow.com/a/44371190/5741172
```
Comment (timche) [inline README.md]: For some reason this whole readme is displayed as a diff. I think there is something wrong with the encoding. Can you please check that it's UTF-8?

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,6 @@
+import { Reducer } from 'redux';
+
+export default function reduceReducer<S>(
+  initialState: S,
+  ...reducers: Array<Reducer<S>>
+): Reducer<S>;
```
Comment (timche) [inline index.d.ts]: Do we need `Reducer` from `redux` as a dependency here? `reduce-reducers` is actually not Redux specific, so we might want to have our own `Reducer` type like `type Reducer<S> = (state: S | undefined, payload: any) => S` if I'm not wrong.

File: README.md (hunk 2 above)
Comment (RumyantsevMichael) [reply README.md]: I don't realize what causes the problem yet

**Comment by lwolle** [comment]
Hey @timche @RumyantsevMichael, I was curious if there are blockers with this issue and if I can offer some help?

**Comment by timche** [comment]
Actually not @lwolle . I just wasn't aware that I should re-review. Anyway, there are still some CI errors to fix before I can review.

**Comment by lwolle** [comment]
Thanks for the fast reply. I'm just waiting for this PR to go through.

**Comment by RumyantsevMichael** [comment]
> Actually not @lwolle . I just wasn't aware that I should re-review. Anyway, there are still some CI errors to fix before I can review.

To be honest, I don't understand what is happening in Travis. Locally both prettier and xo work perfect.

File: index.d.ts (hunk 3 above)
Comment (RumyantsevMichael) [reply index.d.ts]: I fixed that

File: README.md (hunk 1 above)
Comment (RumyantsevMichael) [reply README.md]: Done

**Comment by lwolle** [comment]
Guys, sorry being annoying. But is there any change to get this through as soon as possible? Would you mind if I have a look, or @timche would you mind? I have no experience with the github integrated build pipelines, but would like to support.

**Comment by nikolay-borzov** [comment]
I think the problem is in extra spaces. I guess one can take @RumyantsevMichael's  branch and create his own PR

**Comment by timche** [comment]
@RumyantsevMichael can you please run `yarn format` to format `src/index.js`. This should resolve the issue.

File: README.md [hunk 4]
```diff
@@ -1,8 +1,6 @@
 # reduce-reducers
 
-[![Build Status](https://travis-ci.org/redux-utilities/reduce-reducers.svg?branch=master)](https://travis-ci.org/redux-utilities/reduce-reducers)
-[![npm Version](https://img.shields.io/npm/v/reduce-reducers.svg)](https://www.npmjs.com/package/reduce-reducers)
-[![npm Downloads Monthly](https://img.shields.io/npm/dm/reduce-reducers.svg)](https://www.npmjs.com/package/reduce-reducers)
+[![Build Status](https://travis-ci.org/redux-utilities/reduce-reducers.svg?branch=master)](https://travis-ci.org/redux-utilities/reduce-reducers)[![npm Version](https://img.shields.io/npm/v/reduce-reducers.svg)](https://www.npmjs.com/package/reduce-reducers)[![npm Downloads Monthly](https://img.shields.io/npm/dm/reduce-reducers.svg)](https://www.npmjs.com/package/reduce-reducers)
```
Comment (timche) [inline README.md]: There are still unneeded changes here. Can you please run `git checkout master README.md -p` and revert this.

**Comment by RumyantsevMichael** [comment]
Hmm, interesting. I've installed yarn and now prettier works just like in Travis. But actually it works wrong because now it ignores  the `printWidth` parameter and makes lines longer than 80 characters.

**Comment by RumyantsevMichael** [comment]
@timche, please have a look at it.

**Comment by timche** [comment]
Looks good now, thanks a lot! 👏

**Comment by lwolle** [comment]
Thank you guys! 🏆

**Comment by timche** [comment]
Released in [`v1.0.0`](https://github.com/redux-utilities/reduce-reducers/releases/tag/v1.0.0).

**Comment by RumyantsevMichael** [comment]
Oops, sorry guys. I forgot to add `index.d.ts` into the `files` section of the `package.json` so now it's not in a package.
I will fix it soon.

**Comment by timche** [comment]
@RumyantsevMichael thanks for noticing. Fixed in [`v1.0.1`](https://github.com/redux-utilities/reduce-reducers/releases/tag/v1.0.1)

**Pull request merged**